content/en/docs/Writing policies/preconditions.md

```diff
@@ -15,12 +15,124 @@ For `validate` rules, the use of `patterns` is often preferable since conditiona
 
 When specifying a JMESPath expression in a `preconditions` statement which contains a special character (ex. `/` in the case of Kubernetes annotations), double quote the annotation as a literal string. Escape the double quotes with a backslash character (`\`).
 
-```yaml
+```sh
 {{request.object.spec.template.metadata.annotations.\"foo.k8s.corp.net/bar\"}}
 ```
 
 You may specify multiple statements in the `preconditions` field. The use of multiple `preconditions` statements function as a logical AND statement.
 
+## Any and All Statements
+
+You may further control how `preconditions` are evaluated by nesting the expressions under `any` and/or `all` statements. This gives you further power in building more precise logic for how the rule is triggered. Either or both may be used simultaneously in the same rule with multiple `any` statements also being possible. For each `any`/`all` statement, each block must overall evaluate to TRUE for the precondition to be processed. If any of the `any` / `all` statement blocks does not evaluate to TRUE, `preconditions` will not be satisfied and thus the rule will not be applicable.
```

Comment:
> with multiple `any` statements also being possible

Should this be multiple `any`/`all` statements....

Reply:
This is something I [commented on in this PR](https://github.com/kyverno/kyverno/pull/1791#issuecomment-818709013). Multiple `all` statements don't really make sense because that doesn't address any use case that a single `all` statement couldn't address.

Reply:
Makes sense! Should we state that to avoid the question?

> Either or both may be used simultaneously in the same rule with multiple `any` statements also being possible (multiple `all` statements would be redundant.)

Reply:
That's probably a good idea. I'll send an update.